sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -783,7 +783,7 @@ class SQLContext(@transient val sparkContext: SparkContext)
 
   /** Returns the specified table as a [[DataFrame]]. */
   def table(tableName: String): DataFrame =
-    DataFrame(this, catalog.lookupRelation(Seq(tableName)))
+    DataFrame(this, UnresolvedRelation(Seq(tableName)))
```

Comment:
@rxin this line is the main change. @chenghao-intel thinks we should use `UnresolvedRelation` at here. So, when eager analysis is off, the `DataFrame.queryExecution.logical` will not return a resolved relation.